agent/rpc/middleware/rate_limit_mappings.go

```diff
@@ -0,0 +1,144 @@
+	"ACL.AuthMethodDelete":  rate.OperationTypeWrite,
+	"ACL.AuthMethodList":    rate.OperationTypeRead,
+	"ACL.AuthMethodRead":    rate.OperationTypeRead,
+	"ACL.AuthMethodSet":     rate.OperationTypeWrite,
+	"ACL.Authorize":         rate.OperationTypeRead,
+	"ACL.BindingRuleDelete": rate.OperationTypeWrite,
+	"ACL.BindingRuleList":   rate.OperationTypeRead,
+	"ACL.BindingRuleRead":   rate.OperationTypeRead,
+	"ACL.BindingRuleSet":    rate.OperationTypeWrite,
+	"ACL.BootstrapTokens":   rate.OperationTypeRead,
+	"ACL.Login":             rate.OperationTypeWrite,
+	"ACL.Logout":            rate.OperationTypeWrite,
+	"ACL.PolicyBatchRead":   rate.OperationTypeRead,
+	"ACL.PolicyDelete":      rate.OperationTypeWrite,
+	"ACL.PolicyList":        rate.OperationTypeRead,
+	"ACL.PolicyRead":        rate.OperationTypeRead,
+	"ACL.PolicyResolve":     rate.OperationTypeRead,
+	"ACL.PolicySet":         rate.OperationTypeWrite,
+	"ACL.ReplicationStatus": rate.OperationTypeRead,
+	"ACL.RoleBatchRead":     rate.OperationTypeRead,
+	"ACL.RoleDelete":        rate.OperationTypeWrite,
+	"ACL.RoleList":          rate.OperationTypeRead,
+	"ACL.RoleRead":          rate.OperationTypeRead,
+	"ACL.RoleResolve":       rate.OperationTypeRead,
+	"ACL.RoleSet":           rate.OperationTypeWrite,
+	"ACL.TokenBatchRead":    rate.OperationTypeRead,
+	"ACL.TokenClone":        rate.OperationTypeRead,
+	"ACL.TokenDelete":       rate.OperationTypeWrite,
+	"ACL.TokenList":         rate.OperationTypeRead,
+	"ACL.TokenRead":         rate.OperationTypeRead,
+	"ACL.TokenSet":          rate.OperationTypeWrite,
+
+	"AutoConfig.InitialConfiguration": rate.OperationTypeRead,
+
+	"AutoEncrypt.Sign": rate.OperationTypeWrite,
+
+	"Catalog.Deregister":          rate.OperationTypeWrite,
+	"Catalog.GatewayServices":     rate.OperationTypeRead,
+	"Catalog.ListDatacenters":     rate.OperationTypeRead,
+	"Catalog.ListNodes":           rate.OperationTypeRead,
+	"Catalog.ListServices":        rate.OperationTypeRead,
+	"Catalog.NodeServiceList":     rate.OperationTypeRead,
+	"Catalog.NodeServices":        rate.OperationTypeRead,
+	"Catalog.Register":            rate.OperationTypeWrite,
+	"Catalog.ServiceList":         rate.OperationTypeRead,
+	"Catalog.ServiceNodes":        rate.OperationTypeRead,
+	"Catalog.VirtualIPForService": rate.OperationTypeRead,
+
+	"ConfigEntry.Apply":                rate.OperationTypeWrite,
+	"ConfigEntry.Delete":               rate.OperationTypeWrite,
+	"ConfigEntry.Get":                  rate.OperationTypeRead,
+	"ConfigEntry.List":                 rate.OperationTypeRead,
+	"ConfigEntry.ListAll":              rate.OperationTypeRead,
+	"ConfigEntry.ResolveServiceConfig": rate.OperationTypeRead,
+
+	"ConnectCA.ConfigurationGet": rate.OperationTypeRead,
+	"ConnectCA.ConfigurationSet": rate.OperationTypeWrite,
+	"ConnectCA.Roots":            rate.OperationTypeRead,
+	"ConnectCA.Sign":             rate.OperationTypeWrite,
+	"ConnectCA.SignIntermediate": rate.OperationTypeWrite,
```

Comment:
Not for the scope of this PR, but the `ConnectCA.Roots`, `ConnectCA.Sign`, `ConnectCA.SignIntermediate` are used both to auto generate certs for the servers and to sign leaf for clients. 

I'm wondering if we need to find a way to bypass the call from the server and rate limit the calls to sign client certs somehow 🤔

Reply:
Not actionable in this PR.

Reply:
Added a task (NET-1893) to track this